scipy/stats/_unuran/unuran_wrapper.pyx.templ

```diff
@@ -918,6 +924,428 @@ cdef class TransformedDensityRejection(Method):
+        adaptive Gauss-Lobatto integration. Thus a table of CDF points is stored to keep
+        the number of evaluations of the PDF minimal. Usually this table is discarded when
+        the setup is completed. If ``keep_cdf`` is TRUE, then this table is kept and the
+        approximated CDF can be evaluated by calling the `cdf` method. Default is ``False``.
+    random_state : {None, int, `numpy.random.Generator`,
+                        `numpy.random.RandomState`}, optional
+
+        A NumPy random number generator or seed for the underlying NumPy random
+        number generator used to generate the stream of uniform random numbers.
+        If `random_state` is None (or `np.random`), the `numpy.random.RandomState`
+        singleton is used.
+        If `random_state` is an int, a new ``RandomState`` instance is used,
+        seeded with `random_state`.
+        If `random_state` is already a ``Generator`` or ``RandomState`` instance then
+        that instance is used.
+
+    References
+    ----------
+    .. [1] G. Derflinger, W. Hörmann, and J. Leydold (2008). Numerical inversion when
+           only the density function is known, Research Report Series of the Department
+           of Statistics and Mathematics 78, WU Wien, Augasse 2-6, A-1090 Wien, Austria,
+           http://epub.wu.ac.at/english/
+    .. [2] UNU.RAN reference manual, Section 5.3.12,
+           "PINV – Polynomial interpolation based INVersion of CDF",
+           https://statmath.wu.ac.at/software/unuran/doc/unuran.html#PINV
+
+    Examples
+    --------
+    >>> from scipy.stats import NumericalInversePolynomial
+    >>> from scipy.stats import norm
+
+    To create a generator to sample from the standard normal distribution, do:
+
+    >>> class StandardNormal:
+    ...    def pdf(self, x):
+    ...        return np.exp(-0.5 * x*x)
+    ... 
+    >>> dist = StandardNormal()
+    >>> urng = np.random.default_rng()
+    >>> rng = NumericalInversePolynomial(dist, random_state=urng)
+
+    Once a generator is created, samples can be drawn from the distribution by calling
+    the `rvs` method:
+
+    >>> rng.rvs()
+    -1.5244996276336318
+
+    As this is an invertion method, the approximated PPF can be evaluated by calling
+    the `ppf` method:
+
+    >>> rng.ppf(0.975)
+    1.9599639857012559
+    >>> norm.ppf(0.975)
+    1.959963984540054
+
+    We can see how well the generator approximates the standard normal distribution by
+    plotting the error between approximated PPF and exact PPF of the distribution:
+
+    >>> import matplotlib.pyplot as plt
+    >>> u = np.linspace(0.01, 0.99, 1000)
+    >>> approxppf = rng.ppf(u)
+    >>> exactppf = norm.ppf(u)
+    >>> error = np.abs(exactppf - approxppf)
+    >>> plt.plot(u, error)
+    >>> plt.xlabel('u')
+    >>> plt.ylabel('error')
+    >>> plt.title('Error between exact and approximated PPF')
+    >>> plt.show()
+
+    During setup, the CDF of the distribution is computed numerically but it is
```

Comment:
we could think about moving this to the tutorial, since it is not that important

Reply:
Done.